apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/OrderRow/index.tsx

```diff
@@ -173,8 +176,17 @@ export function OrderRow({
 
   const showCancellationModal = orderActions.getShowCancellationModal(order)
 
+  const [hasValidPendingPermit, setHasValidPendingPermit] = useState<boolean | undefined>(undefined)
+
+  // TODO: do this properly! Maybe an atom & updater on modules/permit and consume it from there?
+  useEffect(() => {
+    if (checkHasValidPendingPermit && orderParams.hasEnoughAllowance === false) {
+      checkHasValidPendingPermit(order).then(setHasValidPendingPermit)
+    }
+  }, [checkHasValidPendingPermit, order, orderParams.hasEnoughAllowance])
```

Comment:
At this point this is checked only once per load.
Working on making this an interval.